tests/src/Simple/PInvoke/PInvoke.cs

```diff
@@ -97,44 +100,64 @@ private static void TestString()
+        }
+
         private static void TestSafeHandle()
         {
             Console.WriteLine("Testing marshalling SafeHandle");
-            String lpFileName = "A.txt";
-            uint dwDesiredAccess = 0x40000000; // GENERIC_WRITE
-            uint dwShareMode = 0x00000002; // FILE_SHARE_WRITE
+            String lpFileName = "PInvoke.cs";
+
+#if Windows_NT
+            uint dwDesiredAccess = 0x80000000; // GENERIC_READ
+            uint dwShareMode = 0x00000001; // FILE_SHARE_READ
             IntPtr lpSecurityAttributes = IntPtr.Zero;
-            uint dwCreationDisposition = 2;// CREATE_ALWAYS;
-            uint dwFlagsAndAttributes = 0x04000000; //FILE_FLAG_DELETE_ON_CLOSE
+            uint dwCreationDisposition = 3;// OPEN_EXISTING;
+            uint dwFlagsAndAttributes = 0x08000000; //FILE_FLAG_ATTRIBUTE_NORMAL
             IntPtr hTemplateFile = IntPtr.Zero;
 
             //create the handle
             SafeFileHandle hnd = SafeFileHandle.CreateFile(lpFileName, dwDesiredAccess, dwShareMode,
                 lpSecurityAttributes, dwCreationDisposition,
                 dwFlagsAndAttributes, hTemplateFile);
-
+#else
+            uint attr = 0x0;  //O_RDONLY
+            int permission = 4; // R for other
+            //create the handle
+            SafeFileHandle hnd = SafeFileHandle.Open(lpFileName, attr, permission);
+                        
+#endif
             IntPtr hndIntPtr = hnd.DangerousGetHandle(); //get the IntPtr associated with hnd
             int intVal =  hndIntPtr.ToInt32(); //return the 32-bit value associated with hnd
 
-            ThrowIfNotEquals(true, SafeHandleTest(hnd, intVal), "Ansi String marshalling failed.");
+            ThrowIfNotEquals(true, SafeHandleTest(hnd, intVal), "SafeHandle marshalling failed.");
         }
     }
 
     public class SafeFileHandle : SafeHandle //SafeHandle subclass
     {
 
-        [DllImport("kernel32", SetLastError = true)]
-        private static extern bool CloseHandle(IntPtr handle);
-
 
+#if Windows_NT
         //each SafeHandle subclass will expose a static method for instance creation
-        [DllImport("kernel32", EntryPoint = "CreateFileW", SetLastError = true, CharSet =CharSet.Unicode)]
+        [DllImport("*", EntryPoint = "CreateFileW", SetLastError = true, CharSet = CharSet.Unicode)]
         public static extern SafeFileHandle CreateFile(String lpFileName,
                                                 uint dwDesiredAccess, uint dwShareMode,
                                                 IntPtr lpSecurityAttributes, uint dwCreationDisposition,
                                                 uint dwFlagsAndAttributes, IntPtr hTemplateFile);
-
-        //default constructor which just calls the base class constructor
+        [DllImport("*", SetLastError = true)]
+        public static extern bool CloseHandle(IntPtr handle);
+#else
+        [DllImport("*", SetLastError = true, CharSet = CharSet.Unicode, EntryPoint="SystemNative_Open")]
```

Comment:
Why not to just a new C method to your library to keep this simple and OS independent? This does not have to be testing file handles.